pyanaconda/modules/timezone/installation.py

```diff
@@ -0,0 +1,85 @@
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import os.path
+from pyanaconda.timezone import is_valid_timezone, write_timezone_config
+from pyanaconda import ntp
+from pyanaconda.modules.common.task import Task
+from pyanaconda.modules.common.constants.services import TIMEZONE
+from pyanaconda.anaconda_loggers import get_module_logger
+
+__all__ = ["ConfigureTimezoneTask"]
+
+log = get_module_logger(__name__)
+
+
+class ConfigureTimezoneTask(Task):
+    """Installation task for timezone configuration."""
+    def __init__(self, sysroot, timezone):
+        """Create a new task,
+        :param sysroot: a path to the root of the installed system
+        :param str timezone: time zone to set
+        """
+        super().__init__()
+        self._sysroot = sysroot
+        self.timezone = timezone
+
+    @property
+    def name(self):
+        return "Configure time zone"
+
+    def run(self):
+        # get the DBus proxies
+        timezone_proxy = TIMEZONE.get_proxy()
+
+        # write out timezone configuration
+        kickstart_timezone = timezone_proxy.Timezone
+
+        # write out timezone configuration
+        # kickstart_timezone = self.timezone
+
+        if not is_valid_timezone(kickstart_timezone):
+            # this should never happen, but for pity's sake
+            log.warning("Timezone %s set in kickstart is not valid, "
+                        "falling back to default (America/New_York).", kickstart_timezone)
+            timezone_proxy.SetTimezone("America/New_York")
+
+        write_timezone_config(timezone_proxy, self._sysroot)
+
+        # write out NTP configuration (if set) and --nontp is not used
+        kickstart_ntp_servers = timezone_proxy.NTPServers
+
+        if timezone_proxy.NTPEnabled and kickstart_ntp_servers:
+            chronyd_conf_path = os.path.normpath(self._sysroot + ntp.NTP_CONFIG_FILE)
+            pools, servers = ntp.internal_to_pools_and_servers(kickstart_ntp_servers)
+            if os.path.exists(chronyd_conf_path):
+                log.debug("Modifying installed chrony configuration")
+                try:
+                    ntp.save_servers_to_config(pools, servers, conf_file_path=chronyd_conf_path)
+                except ntp.NTPconfigError as ntperr:
+                    log.warning("Failed to save NTP configuration: %s", ntperr)
```

Comment:
Isn't this more an error than warning? We are not able to write what user asked for.

Reply:
Committer of the original code explicitly chose warning instead of error here...

Reply:
Yeah, this is more subjective feeling I guess. You can leave it be.